.github/workflows/pull_request.yml

```diff
@@ -21,3 +21,155 @@ jobs:
+          # depending on whether we are building locally or in a container, add a maker to the name
+          if [[ "${{ matrix.build-type }}" == 'local' ]]; then
+            ARTIFACT_NAME="${ARTIFACT_NAME}-local"
+          fi
+          if [[ "${{ matrix.build-compiler }}" == '1' ]]; then
+            ARTIFACT_NAME="${ARTIFACT_NAME}-hostbuild"
+          fi
+          # artifacts need a unique name so we suffix with the matrix arch(s)
+          if [[ ! -z "${{ matrix.arch }}" ]]; then
+            ARTIFACT_NAME="${ARTIFACT_NAME}-$(echo ${{ matrix.arch }} | tr ',' '-')"
+          fi
+          ARTIFACT_NAME="${ARTIFACT_NAME}${ARTIFACT_EXT}"
+
+          # There is no way to prevent even a single-file artifact from being zipped:
+          # https://github.com/actions/upload-artifact?tab=readme-ov-file#zip-archives
+          # so the actual artifact download will look like:
+          # swift-6.1-RELEASE_android-0.1-x86_64.artifactbundle.tar.gz.zip
+          echo "artifact-name=${ARTIFACT_NAME}" >> $GITHUB_OUTPUT
+      - name: Upload SDK artifactbundle
+        uses: actions/upload-artifact@v4
+        with:
+          compression-level: 0
+          name: ${{ steps.info.outputs.artifact-name }}
+          path: ${{ steps.info.outputs.artifact-path }}
+      - name: Cleanup
+        run: |
+          # need to free up some space or else when installing we get: No space left on device
+          df -h
+          rm -rf ${WORKDIR}/{build,source}
+          sudo docker image prune --all --force
+          sudo docker builder prune -a
+          df -h
+      - name: Install artifactbundle
+        shell: bash
+        run: |
+          set -ex
+          ${{ steps.info.outputs.swift-path }} sdk install ${{ steps.info.outputs.artifact-path }}
+          ${{ steps.info.outputs.swift-path }} sdk configure --show-configuration $(${{ steps.info.outputs.swift-path }} sdk list | head -n 1) ${{ steps.info.outputs.sdk-id }}
+          # recent releases require that ANDROID_NDK_ROOT *not* be set
+          # see https://github.com/swiftlang/swift-driver/pull/1879
+          echo "ANDROID_NDK_ROOT=" >> $GITHUB_ENV
+
+      - name: Create Demo Project
+        run: |
+          cd ${{ runner.temp }}
+          mkdir DemoProject
+          cd DemoProject
+          ${{ steps.info.outputs.swift-path }} --version
+          ${{ steps.info.outputs.swift-path }} package init
+          echo 'import Foundation' >> Sources/DemoProject/DemoProject.swift
+          echo 'import FoundationEssentials' >> Sources/DemoProject/DemoProject.swift
+          echo 'import FoundationXML' >> Sources/DemoProject/DemoProject.swift
+          echo 'import FoundationNetworking' >> Sources/DemoProject/DemoProject.swift
+          echo 'import Dispatch' >> Sources/DemoProject/DemoProject.swift
+          echo 'import Android' >> Sources/DemoProject/DemoProject.swift
+      - name: Test Demo Project on Android
+        uses: skiptools/swift-android-action@main
```

Comment:
What are we using this action for?

Reply:
Nothing, this entire workflow change is just a demo. We're looking for guidance on how you want to do it.